Attifunel process req test description added (#679)
eclipse-score/process_description: process/process_areas/verification/guidance/verification_methods.rst

   Following derivation techniques are explained

   * :ref:`Requirements analysis <ver_req_anal>`
   * :ref:`Boundary Values <ver_boundary>`
   * :ref:`Equivalence Classes <ver_equivalence>`
   * :ref:`Fuzzy Testing <ver_fuzzy>`

.. _ver_req_anal:

Requirements analysis
"""""""""""""""""""""

Requirements analysis is a testing technique where tests are designed based on the

> requirements assigned to the software element under test (component, module, unit etc.).

Reviewer: ```suggestion
Requirements analysis is a test derivation technique where tests are designed based on the
requirements assigned to the software under test (unit, component, feature, etc.).
```

Author: I would still keep "SW element" as it's the most generic definition in ISO 26262 of any hierarchical piece of SW

Reviewer: Agree, but I would still see it as a test derivation technique.

Author: Ok added "derivation" and fixed "feature" etc.

Author: I think we can resolve this

If available, the verification method defined for each requirement should be used as
the basis of the test cases design.

Requirements should always be tested  against their "nominal" behaviour: i.e. "if input
x = A, the output y of SW component X shall be equal to b". This leads to a test case
injection value x = A and verifying that output y = B. "Negative" testing technique should
be also considered: i.e. testing what happens when input x = B (if not already specified

> in other requirements).

Reviewer: ```suggestion
Requirements should always be tested  against their "nominal" behaviour: i.e. "if input
x = A, the output y of SW component Z shall be equal to B". This leads to a test case
injection value x = A and verifying that output y = B. "Negative" testing technique should
be also considered: i.e. testing what happens when input x = B (if not already specified
in other requirements).
```

Author: I would also change the negative case input to something different than B...

Reviewer: Agree.

Author: I think this is solved

> Note that requirement-based derived test case may also cover other techniques and derivation methods

Reviewer 2: Thats not an "may"! It depends on the ASIL Level according to the ISO Norm. For ASIL B the "Generation and analysis of equivalence classes" and "Analysis of boundary values" are highly recommended (what means mandatory). Fault injection is for ASIL C.

Author: I just wanted to underline that, incidentally, a req. bases test case could also already cover other techniques mentioned in this document. No reference to ISO 26262 requirements on testing here was meant. Agreed with P. Ahmann

Reviewer: Agreed with @attifunel as this is explained in other section.

mentioned here, such as Boundary Values, Equivalence Classes, Fuzzy Testing, Interface

> Testing and Fault Injection.

Reviewer: ```suggestion
Note that requirement-based testing derivation is typically covered by other techniques
and methods mentioned here, such as Boundary Values, Equivalence Classes, and Fuzzy Testing.
```

Interface testing is verification method, but according to the iso not a derivation technique. Same holds true for Fault injection.

Author: Mhmhm... I have this opinion that ISO 26262 makes a bit of mixed bag there by mixing "derivation methods" and "techniques". I wanted to stay generic speaking of "techniques" without derivation, but maybe this causes confusion?

Reviewer: My point of concern is where it says `typically covered by` do you mean it gets `extended by` or `enhanced by`? With the current wording it sounds much more like being included in the other techniques.

Author: Changed to "Note that requirement-based dervied test case may also cover other techniques". Agree?
